node/bft/src/primary.rs

```diff
@@ -768,6 +787,47 @@ impl<N: Network> Primary<N> {
         // Inserts the missing transmissions into the workers.
         self.insert_missing_transmissions_into_workers(peer_ip, missing_transmissions.into_iter())?;
 
+        // Ensure the transaction doesn't bring the proposal above the spend limit.
+        let block_height = self.ledger.latest_block_height() + 1;
+        if N::CONSENSUS_VERSION(block_height)? >= ConsensusVersion::V4 {
```

Comment:
Note that the Consensus/BFT modules do not care about block heights and only care about rounds. 

This means different validators may be at same round, but different block height if one of the validators have not received enough certificates to advance their ledger. This means they may process incoming proposals differently, but neither are technically wrong from their point of view.

Reply:
Intuitively, this likely won't be an issue if the validator from behind syncs up with other certificates and catches their ledger up before processing the incoming proposal.

Reply:
Let me check my understanding: the edge case is a straggling node processing transmissions into a batch without limiting spend when the limit is already enforced by other validators? Eventually all nodes should process the same subdag into a block at the same height so unless there is a quorum of nodes behind (impossible by definition?) there's no chance of a spend limit going unenforced because of asynchrony during the consensus upgrade? 

In other words, despite rounds not matching, block height is still dependable as a trigger for new logic in these modules? 🤔

Reply:
Perhaps we could use a lookback approach for how (the heights of) blocks determine the consensus version active at each round, similarly to how blocks determine the active (= lookback) committee at each round.